### ISSUE TITLE
The MSBuild target AssignLinkMetadata is missing the Resource item group

### ISSUE THREAD

**Issue by matherm-aboehm** [dotnet/wpf#7892]
### Description

I have a shared project with XAML files and where some image files were added to a subdirectory "Resources". I tried to embed these images with the "Resource" build action to use them as source for BitmapImages. I've done this by manually editing the .projitems file and changing the "Content" tags to "Resource" for the image files.
So far this works for the WPF designer: You can chose the images from the dropdown in the property grid and then it will be shown in the preview area of the WPF designer.
But when you run the app, it will throw an IOException with the message that the "Resource 'resources/xx' cannot be found".

### Reproduction Steps

- create a new WPF-App project
- create a new shared project
- add a reference to the shared project from the WPF-App project
- add a WPF UserControl to the shared project
- add an image to a subdirectory of the shared project
- open the XAML file of the new UserControl in the editor/designer
- add a Image tag to the code with the Source property set to the image from the subdirectory
- see the image in the preview area
- add this UserControl into the MainWindow.xaml of the WPF-App project.
- compile and run the app
- the IOException is thrown

### Expected behavior

WPF app shows the image instead of throwing an exception.

### Actual behavior

The app throws an exception of type System.IO.IOException with the message about the missing resource.

### Known Workarounds

You can define and schedule your own MSBuild target in each and every project which references the shared project, so that it includes the missing tasks for Resource items.

### Configuration

- .NET Framework 4.8
- Windows 10 Pro 22H2
- x64
- No, it is not specific to this configuration.

### Other information

When you look into the ".g.ressources" item in the compiled assembly you can see the image file, but without any folder name.
This is caused by the missing tasks of the MSBuild target "AssignLinkMetadata" in "Microsoft.Common.CurrentVersion.targets". There are tasks for Page and ApplicationDefinition items, but not for Resource items.

Was that intentional or just oversight? I can't find any documentation about it.

**Comment by singhashish-wpf** [comment]
@matherm-aboehm Could you please help to share a sample repro?

**Comment by matherm-aboehm** [comment]
I just created the sample repo here: [WpfIssue7892](https://github.com/matherm-aboehm/WpfIssue7892)

Open the MyUserControl1.xaml from SharedProject1, to see the image in WPF designer.
If you want to see the runtime exception, just compile and run the project unmodified.
To activate the workaround, open the csproj file in editor and uncomment last Target definition.

**Comment by Kuldeep-MS** [comment]
@matherm-aboehm - The sample application you provided is designed for .NET Framework, but I have found that the same issue can also be replicated in .NET Core. Upon closer inspection, I have discovered that if we use the image tag in the following way: ```<Image Source="/assignlink;component/IconFile_256x.png" ImageFailed="Image_ImageFailed"/>```, without including the `Resource` folder name, even though the `IconFile_256x.png` is located in the `Resource` folder, there will be no exceptions. I believe you are already aware of this.

Regarding your question "Was that intentional or just oversight?", I cannot provide a definitive answer at this time. Additionally, based on the information you have provided, if you believe that there is a missing task in `Microsoft.Common.CurrentVersion.targets`, you may want to consider opening an issue with `dotnet/msbuild` and close this issue.

**Comment by matherm-aboehm** [comment]
Yes, I'm aware of this behavior, because this is what I initially described here.
But there are some problems writing the image tag like this:
1. This only works because of the missing task in `Microsoft.Common.CurrentVersion.targets`, when this changes it will be broken again.
2. You lose support from WPF designer completely. So, you can't see the image in editor and get various design time errors.
3. You can only use unique file names for the images. This means, you can't use folders as namespaces, but you can for non-shared projects.

Initially I was about to report the issue on `dotnet/msbuild` but then, when i got the choices for the issue type, I saw the following message:
"Please open issues relating to WPF in `dotnet/wpf`, including build issues specific to WPF."

So I thought as this is a build issue specific to WPF it fits best here.
Should I re-post it from scratch under `dotnet/msbuild` or can I just link it there?
